pkg/autodiscovery/listeners/common_test.go

```diff
@@ -250,6 +250,8 @@ func TestGetPrometheusIncludeAnnotations(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			config.Datadog.Set("prometheus_scrape.checks", tt.config)
+			defer config.Datadog.Set("prometheus_scrape.checks", nil)
+
```

Comment:
Would it be safer or more future-proof to store the previous value ? Just in case it happens to not be `nil` in some cases ?

```suggestion
            originalChecks := []*types.PrometheusCheck{}
            err := config.Datadog.UnmarshalKey("prometheus_scrape.checks", &originalChecks)
            assert.NoError(t, err)
			config.Datadog.Set("prometheus_scrape.checks", tt.config)
			defer config.Datadog.Set("prometheus_scrape.checks", originalChecks)
```

Reply:
Good point, thanks!